eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -941,6 +941,16 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
       goto(NORMAL) using d.copy(channelUpdate = channelUpdate) storing() replying ChannelCommandResponse.Ok
 
     case Event(BroadcastChannelUpdate(reason), d: DATA_NORMAL) =>
+      // We usually handle feerate updates once per block (~10 minutes), but when our remote is a mobile wallet that only
+      // briefly connects and then disconnects, we may never have the opportunity to send our `update_fee`, so we always
+      // send it (if needed) when reconnected.
+      if (reason == Reconnected && d.commitments.localParams.isFunder) {
+        val currentFeeratePerKw = d.commitments.localCommit.spec.feeratePerKw
```

Comment:
Why not reusing `handleCurrentFeerate`? It could become:
```
val currentFeeRate = CurrentFeerates(nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(nodeParams.onChainFeeConf.feeTargets.commitmentBlockTarget)
handleCurrentFeerate(currentFeeRate, d)
```

Reply:
This code unfortunately doesn't work, you can't build back the `CurrentFeerates` that easily, it would require ugly code to call `feeEstimator` for every block target...
And `handleCurrentFeerate` does more than what we do, it handles fundee side and does a state transition which we shouldn't do in that specific case.